lib/react-dom.js

```diff
@@ -198,9 +198,7 @@ declare class SyntheticKeyboardEvent<
   which: number;
 }
 
-declare class SyntheticMouseEvent<
-  +T: EventTarget = EventTarget,
-> extends SyntheticUIEvent<T, MouseEvent> {
+class $SyntheticMouseEvent {
```

Comment:
How about making this extend `SyntheticUIEvent`, take two type parameters, and just pass them through?

e.g.:

```
declare class $SyntheticMouseEvent<+T: EventTarget, NativeEvent> extends SyntheticUIEvent<T, MouseEvent> { ... }
```

and then below:

```
declare class SyntheticMouseEvent<+T: EventTarget = EventTarget> extends $SyntheticMouseEvent<T, MouseEvent> { ... }
```

And similar for the others below. I don't think `mixins` is really necessary here.

Reply:
I made changes very nearly to what was suggested. Please, take a second look whenever you'll have free time